cluster/charts/rook/templates/wait-on-resources-job.yaml

```diff
@@ -23,7 +22,7 @@ spec:
           command:
             - ./kubectl
             - get
-            {{- if printf "%s.%s" .Capabilities.KubeVersion.Major .Capabilities.KubeVersion.Minor | semverCompare ">=1.7" }}
+            {{- if .Capabilities.APIVersions.Has "apiextensions.k8s.io/v1beta1" }}
```

Comment:
Nice find @TimJones, this looks like it will be much more reliable than the surprising number of malformed semvers there are out there for kubernetes deployments!  A couple questions:
* did you find any documentation on this function? i'd like to read about it more
* will this still work when CRDs go to stable?  we are checking for `v1beta1` specifically, if the version gets incremented, will this check still work?

Reply:
@jbw976 Documentation can be found [here](https://docs.helm.sh/chart_template_guide/#built-in-objects). It was a feature released in [Helm 2.2](https://github.com/kubernetes/helm/releases/tag/v2.2.0)  

In regards to API version availability, this should work even when CRDs are stable, until the `v1beta1` is fully removed rather than just deprecated (see Deployments working fine for `extentions/v1beta1`, `extentions/v1beta2` and `apps/v1` in K8S 1.9). Even if not, the check can simply be updated to something like: 

```yaml
if or ( .Capabilities.APIVersions.Has "apiextensions.k8s.io/v1beta1" ) ( .Capabilities.APIVersions.Has "whatever.stable.kubernetes.io/v1" )
```

Reply:
That sounds good to me @TimJones since the older API versions are not immediately removed when a new one is released.  Thanks for the explanation and contribution!